src/lib/eliom_shared_content.eliom

```diff
@@ -0,0 +1,418 @@
+
+  end
+
+  module R = struct
+
+    (* Same as the HTML version, with Html5 -> SVG and `HTML5 ->
+       `SVG. We can in principle functorize, but probably not worth
+       the trouble. Make sure they stay synced! *)
+    let node s =
+      let e =
+        local_value s |>
+        Eliom_content_core.Svg.D.toelt |>
+        Eliom_content_core.Xml.make_request_node ~reset:false
+      and synced = React.S.synced s in
+      let _ = {unit{
+        let s =
+          %s >|= (fun s ->
+            Eliom_content_core.Svg.
+              (Id.create_request_elt s ~reset:false |> D.toelt) |>
+            Eliom_client.rebuild_node' `SVG)
+        in
+        let f =
+          let replace e' e =
+            let f p = Dom.replaceChild p e' e in
+            Js.Opt.iter (e##parentNode) f |> ignore
+          and e = Eliom_client.rebuild_node' `SVG %e in
+          fun e' ->
+            replace e' e;
+            React.S.diff replace s |> ignore
+        in
+        if %synced then
+          React.(S.changes s |> E.once |> E.map f) |> ignore
+        else
+          f (React.S.value s) |> ignore
+      }} in
+      e |> Eliom_content_core.Svg.D.tot
+
+    include Eliom_content_core.Svg.Make(Xml)(Wrapped_functions)
+
+  end
+
+end
+
+module Html5 = struct
+
+  module Wrapped_functions :
+
+    Html5_sigs.Wrapped_functions
+    with type (-'a, 'b) ft = ('a, 'b) Xml.W.ft =
+
+  struct
+
+    type (-'a, 'b) ft = ('a, 'b) Xml.W.ft
+
+    let string_of_big_variant () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_big_variant }}
+
+    let string_of_bool () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_bool }}
+
+    let string_of_character () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_character }}
+
+    let string_of_input_type () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_input_type }}
+
+    let string_of_linktypes () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_linktypes }}
+
+    let string_of_mediadesc () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_mediadesc }}
+
+    let string_of_multilength () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_multilength }}
+
+    let string_of_multilengths () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_multilengths }}
+
+    let string_of_numbers () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_numbers }}
+
+    let string_of_sandbox () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_sandbox }}
+
+    let string_of_sizes () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_sizes }}
+
+    let string_of_step () =
+      {shared#{ Html5_f.Wrapped_functions.string_of_step }}
+
+    let unoption_string () =
+      {shared#{ Html5_f.Wrapped_functions.unoption_string }}
+
+  end
+
+  module R = struct
+
+    (* Same as the SVG version, with Svg -> Html5 and `SVG ->
+       `HTML5. We can in principle functorize, but probably not worth
+       the trouble. Make sure they stay synced! *)
```

Comment:
100% chance someone is going to break that. Functorize it. It's not even really hard and it's shorter.